tests/unit/apps_planning/test_auxiliary_subordinate_plans.py

```diff
@@ -0,0 +1,106 @@
+#  Copyright 2024 Canonical Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Tests for AuxiliarySubordinateApplication generating plans."""
+import pytest
+
+from cou.apps.auxiliary_subordinate import (
+    AuxiliarySubordinateApplication,
+    OvnSubordinate,
+)
+from cou.utils.juju_utils import Unit
+from cou.utils.openstack import OpenStackRelease
+from tests.unit.utils import dedent_plan, generate_cou_machine
+
+
+@pytest.mark.parametrize(
+    "app_name, channel, version",
+    [
+        ("hacluster", "2.4/stable", "2.4"),
+        ("mysql-router", "8.0/stable", "8.0"),
+        ("ceph-dashboard", "octopus/stable", "17.0.1"),
+    ],
+)
+def test_auxiliary_application_upgrade_plan(app_name, channel, version, model):
```

Comment:
maybe `test_auxiliary_application_refresh_plan`, as this is only an in-channel refresh